File: ModelInterfacesClassLibrary/interfaces/asp/aspnetroleView/aspnetroleViewPage.interface.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;
using Newtonsoft.Json.Serialization;
using ModelInterfacesClassLibrary.interfaces.asp.aspnetroleView;

namespace ModelInterfacesClassLibrary.interfaces.asp.aspnetroleView {
    public interface IaspnetroleViewPage
    {
        [JsonProperty(PropertyName = "page")]
        int page { get; set; }

        [JsonProperty(PropertyName = "pagesize")]
        int pagesize { get; set; }

        [JsonProperty(PropertyName = "pagecount")]
        int pagecount { get; set; }

        [JsonProperty(PropertyName = "total")]
        int total { get; set; }

        [JsonProperty(PropertyName = "items")]
        IList<IAspnetroleView> items { get; set; }
    }
}

File: AngularMatPhonebookSolution/projects/asp-interfaces/src/lib/components/aspnetrole-view/interfaces/aspnetrole-view-page.interface.ts
import { IaspnetroleView } from './aspnetrole-view.interface';

export interface IaspnetroleViewPage {
        page : number; // int
        pagesize : number; // int
        pagecount : number; // int
        total : number; // int
        items ?: IaspnetroleView[];
}


